test/unit/lib/TopStoriesFeed.test.js

```diff
@@ -1460,4 +1460,38 @@ describe("Top Stories Feed", () => {
     assert.calledOnce(instance.uninit);
     assert.calledOnce(instance.init);
   });
+  describe("#layout", () => {
+    it("should call dispatchLayoutUpdate from loadCachedData and fetchStories", async () => {
+      instance.stories_endpoint = "stories-endpoint";
+      let fetchStub = globals.sandbox.stub();
+
+      const response = {
+        "layout": [1, 2],
+      };
+      globals.set("fetch", fetchStub);
+      fetchStub.resolves({ok: true, status: 200, json: () => Promise.resolve(response)});
+      sinon.spy(instance, "dispatchLayoutUpdate");
+      instance.cache.get = () => ({stories: {layout: [2, 3]}});
+
+      await instance.loadCachedData();
+      assert.calledOnce(instance.dispatchLayoutUpdate);
+      const firstCallArgs = instance.dispatchLayoutUpdate.firstCall.args;
+      assert.equal(firstCallArgs[0][0], 2);
+      assert.equal(firstCallArgs[0][1], 3);
+
+      await instance.fetchStories();
+      assert.calledTwice(instance.dispatchLayoutUpdate);
+      const secondCallArgs = instance.dispatchLayoutUpdate.secondCall.args;
+      assert.equal(secondCallArgs[0][0], 1);
+      assert.equal(secondCallArgs[0][1], 2);
+    });
+    it("should call dispatch from dispatchLayoutUpdate if preffed on and with available data", () => {
```

Comment:
"if preffed on and" is probably no longer necessary here.  :-)

Reply:
Do we want another test here that makes sure that if the data is unavailable, that nothing gets dispatched?

Reply:
k

Reply:
Thanks.